File: 2025_1/stack.py
class Stack:
    def __init__(self, max_size=100):
        self.data =[]
        self.max_size = max_size
        
    def push(self, value):
        if len(self.data) >= self.max_size:
            print("stack is full")
        else:
            self.data.append(value)
        
    def pop(self):
        if len(self.data) == 0:
            print("stack is empty")
        else:
            return self.data.pop()
        
    
    def is_empty(self):
        return len(self.data) == 0
        
    def is_full(self):
        return len(self.data) == self.max_size
        
    def display(self):
        if self.is_empty():
            print("stack is empty")
        else:
            print(self.data)

File: 2025_1/stack.c
#include <stdio.h>

#define MAX_SIZE 100

typedef int element;

typedef struct {
    element data[MAX_SIZE];
    int top;
} Stack;


void init(Stack *stk) {
    stk->top = -1;
}

void push(Stack *stk, element value){
    if (stk->top == MAX_SIZE -1) {
        printf("stack is full\n");
    } else {
        stk->data[++(stk->top)] = value;
    }
}

void pop(Stack *stk, element *value){
    if (stk->top == -1) {
        printf("stack is empty\n");
    } else {
        *value = stk->data[(stk->top)--];
    }
}


void isEmpty (Stack *stk) {
    if (stk->top == -1) {
        printf("stack is empty\n");
    }
}

void isFull (Stack *stk) {
    if (stk->top == MAX_SIZE -1) {
        printf("stack is full\n");
    }
}


void display(Stack *stk) {
    if (stk->top == -1) {
        printf("stack is empty\n");
    } 
    for (int i=0; i<=stk->top; i++) {
        printf("%d", stk->data[i]);
    }
    printf("\n");
}